src/groups/mqb/mqbc/mqbc_storageutil.cpp

```diff
@@ -198,13 +280,33 @@ void StorageUtil::updateQueuePrimaryDispatched(
 
     bslmt::LockGuard<bslmt::Mutex> guard(storagesLock);  // LOCK
 
+    AppInfos existingAppInfos;
+    storage->loadVirtualStorageDetails(&existingAppInfos);
+
+    bmqu::Printer<AppInfos> printer2(&existingAppInfos);
+
+    BALL_LOG_INFO << clusterDescription << " Partition [" << partitionId
+                  << "]: Existing queue '" << storage->queueUri()
+                  << "', queueKey: '" << storage->queueKey() << "' "
+                  << printer2 << " in the storage.";
+
+    AppInfos addedAppInfos, removedAppInfos;
+
+    bool hasUpdate = loadUpdatedAppInfos(&addedAppInfos,
+                                         &removedAppInfos,
+                                         *storage,
```

Comment:
I recommend changing the function signature of `loadUpdatedAppInfos` to take in `existingAppInfos` instead of `*storage`. Currently, you are calling `storage->loadVirtualStorageDetails(&existingAppInfos);` twice in quick succession.